### PR TITLE
ISPN-7688 Fix some security tests when running under the IBM JDK

### PR DESCRIPTION
**Pull request by tristantarrant**
- Use IBM specific Kerberos login module options
- Perform a double Kerberos login to workaround an issue with "invalid options"
- Use a keystore generated using the IBM JDK for the vault tests

https://issues.jboss.org/browse/ISPN-7688

### CODE REVIEW

**Comment by danberindei** [comment]
@tristantarrant @slaskawi Any overlap between this and https://github.com/infinispan/infinispan/pull/5036 ?

**Comment by tristantarrant** [comment]
This only fixes the tests: we test against "prepared" keystores which are JDK-specific.

File: server/integration/testsuite/src/test/java/org/infinispan/server/test/client/hotrod/security/HotRodKrbAuthIT.java [hunk 1]
```diff
@@ -48,11 +48,18 @@ public RemoteInfinispanServer getRemoteServer() {
...
+      String krbLogin = isIBMJDK ? "/ibm_jaas_krb_login.conf" : "/jaas_krb_login.conf";
       System.setProperty("java.security.auth.login.config", HotRodKrbAuthIT.class.getResource(krbLogin)
             .getPath());
       System.setProperty("java.security.krb5.conf", HotRodKrbAuthIT.class.getResource("/krb5.conf").getPath());
       LoginContext lc = new LoginContext("HotRodKrbClient", new SimpleLoginHandler(login + "@" + KRB_REALM, password));
+      if (isIBMJDK) {
+         // workaround for IBM JDK: the first negotiation always fails, so let's do a dummy login/logout round.
+         lc.login();
+         lc.logout();
+         lc = new LoginContext("HotRodKrbClient", new SimpleLoginHandler(login + "@" + KRB_REALM, password));
```
Comment (danberindei) [inline server/integration/testsuite/src/test/java/org/infinispan/server/test/client/hotrod/security/HotRodKrbAuthIT.java]: Do we need to write something about this in the FAQ, too?

**Comment by tristantarrant** [comment]
@danberindei I have added a comment to the FAQ

**Comment by danberindei** [comment]
Integrated, thanks Tristan!

**Pull request closed without merging**